tests/third_party/cupy/manipulation_tests/test_basic.py

```diff
@@ -128,8 +155,83 @@ def test_copyto(self, xp, dtype):
     @testing.numpy_cupy_allclose(accept_error=TypeError)
     def test_copyto_where(self, xp, dtype):
         dst = xp.ones(self.dst_shape, dtype=dtype)
-        mask = (testing.shaped_arange(self.dst_shape, xp, dtype) % 2).astype(
-            xp.bool_
+        # TODO: replace with
+        # mask = (testing.shaped_arange(
+        #     self.dst_shape, xp, dtype) % 2).astype(xp.bool_)
+        # once dpnp.reminder() is fully implemented
+        mask = xp.array(
+            testing.shaped_arange(self.dst_shape, numpy, dtype) % 2,
+            dtype=xp.bool_,
         )
         xp.copyto(dst, self.src, where=mask)
         return dst
+
+
+@pytest.mark.parametrize(
+    "casting", ["no", "equiv", "safe", "same_kind", "unsafe"]
+)
+class TestCopytoFromNumpyScalar:
+    @testing.for_all_dtypes_combination(("dtype1", "dtype2"))
+    @testing.numpy_cupy_allclose(accept_error=TypeError)
+    def test_copyto(self, xp, dtype1, dtype2, casting):
+        if casting == "safe":
+            pytest.skip(
+                "NEP50 doesn't work properly in numpy with casting='safe'"
+            )
+
+        dst = xp.zeros((2, 3, 4), dtype=dtype1)
+        src = numpy.array(1, dtype=dtype2)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", numpy.ComplexWarning)
+            xp.copyto(dst, src, casting)
+        return dst
+
+    @testing.for_all_dtypes()
+    @pytest.mark.parametrize(
+        "make_src",
+        [lambda dtype: numpy.array([1], dtype=dtype), lambda dtype: dtype(1)],
+    )
+    @testing.numpy_cupy_allclose(accept_error=TypeError)
+    def test_copyto2(self, xp, make_src, dtype, casting):
+        dst = xp.zeros((2, 3, 4), dtype=dtype)
+        src = make_src(dtype)
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", numpy.ComplexWarning)
+            xp.copyto(dst, src, casting)
+        return dst
+
+    @testing.for_all_dtypes_combination(("dtype1", "dtype2"))
+    @testing.numpy_cupy_allclose(accept_error=TypeError)
+    def test_copyto_where(self, xp, dtype1, dtype2, casting):
+        if casting == "safe":
+            pytest.skip(
+                "NEP50 doesn't work properly in numpy with casting='safe'"
+            )
+
+        shape = (2, 3, 4)
+        dst = xp.ones(shape, dtype=dtype1)
+        src = numpy.array(1, dtype=dtype2)
+        # TODO: replace with
+        # mask = (testing.shaped_arange(shape, xp, dtype1) % 2).astype(xp.bool_)
+        # once dpnp.reminder() is fully implemented
```

Comment:
Now that the new implementation for `dpnp.remainder` is merged, you can modify this part as well.

Reply:
thanks a lot for the reminder!